### ISSUE TITLE
Native linking of iOS project fails with undefined symbols with 2.3.0

### ISSUE THREAD

**Issue by JR-Fire** [CrossGeeks/FirebasePushNotificationPlugin#108]
## Bug Information

Version Number of Plugin: 2.3.0
Device Tested On: n/a
Simulator Tested On: n/a
Version of VS: 2017 15.5.6 (Enterprise) on windows, building iOS with the Xamarin Mac Agent (Xcode 9.2)
Version of Xamarin: 4.8.0.760
Versions of other things you are using:
nothing relevant as I can reproduce with the sample, too

### Steps to reproduce the Behavior
I have a Xamarin PCL and two mobile app project - Android and iOS. Both with working notifications using Plugin.FirebasePushNotification 2.1.3. Both happily building and deploying, etc. After upgrading to the latest plugin version 2.3.0 (b/c Android Oreo), the iOS project no longer builds.

Reproducible with the sample project - I downloaded the sample project and I can reproduce the issue, attaching the build output.

### Expected Behavior
Project to build successfully

### Actual Behavior
Building of iOS project fails with undefined symbols:
2>  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(777,3): error : Native linking failed, undefined symbol: _kFIRServiceFirestore. Please verify that all the necessary frameworks have been referenced and native libraries are properly linked in.
2>  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(777,3): error : Native linking failed, undefined symbol: _kFIRServicePerformance. Please verify that all the necessary frameworks have been referenced and native libraries are properly linked in.
2>  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(777,3): error : Native linking failed. Please review the build log.

### Screenshots
Build output (this one is from the sample project build failing): https://gist.github.com/JR-Fire/e2fa548110980c512d700d45abd57ad8

**Comment by JR-Fire** [comment]
I've been trying to update the plugin for a couple of days now and I'd appreciate any help. I'm able to reproduce the linking failed error with the sample project, too. I've compared our build configuration with the one in the sample project and it's the same (save for signing, of course) - same linker behavior, same architectures.
Thanks in advance!

**Comment by rdelrosario** [comment]
Hi

Did you change the Bundle identifier before building?

I just cloned the repository from scratch and did the following steps:
1. Clean
2. Restore Packages
3. Rebuild
4. Deploy on iPhone

Didn't face any issues. I'm deploying from VS on my PC connected to my Mac.

**Comment by rdelrosario** [comment]
Btw, reset your Mac just in case is a cache issue.

**Comment by JR-Fire** [comment]
Hi,

I'll see about deleting any xamarin agent cache tomorrow as it's a shared
development machine with other teams. Resetting the Mac is probably not an
option but I also think it's unnecessary as we have no other failing builds.

Are you saying I need to change the bundle id of the sample and then try to
build it? I'm not sure how that would affect linking.

Best regards,
JR

**Comment by rdelrosario** [comment]
Yes won't built because won't be able to sign with that bundle identifier if not registered on your Apple Developer account.

**Comment by JR-Fire** [comment]
Ah, no, that's correctly set up - the samples project is set up to
"automatical" (it uses the default profile).
Btw, in the build output it selects a profile and continues with the
building. Also, as I mentioned, if I remove the notifications and the
references, it builds successfully (that's valid for the sample project,
too).

I'll see for any caches later, if you can build it, it's probably not a bug.


On Feb 20, 2018 8:40 PM, "Rendy Del Rosario" <notifications@github.com>
wrote:

Yes won't built because won't be able to sign with that bundle identifier
if not registered on your Apple Developer account.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/CrossGeeks/FirebasePushNotificationPlugin/issues/108#issuecomment-367076323>,
or mute the thread

**Comment by rdelrosario** [comment]
Interesting, I will try to test on other machines to see If I can repo. But on my environment from scratch is working fine.

**Comment by JR-Fire** [comment]
Thank you! Clearing the Xcode cache didn't help. I'm also in the process of setting up another mac for building (but it's gonna take a while :) ) I'll report back if a clean install of XCode & Xamarin works.

**Comment by JR-Fire** [comment]
Hi,
We've setup a second Mac to build - brand new Xcode + Visual Studio + Xamarin install, the same linking errors reproduce:

5>"X:\mQES\_NET_trunk\src\DAIS\DAIS.PCL\DAIS.PCL.mQES.App\DAIS.PCL.mQES.App.iOS\DAIS.PCL.mQES.App.iOS.csproj" (Rebuild target) (1) ->
5>(_CompileToNative target) -> 
5>  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(777,3): error : Native linking failed, undefined symbol: _kFIRServiceFirestore. Please verify that all the necessary frameworks have been referenced and native libraries are properly linked in.
5>  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(777,3): error : Native linking failed, undefined symbol: _kFIRServicePerformance. Please verify that all the necessary frameworks have been referenced and native libraries are properly linked in.
5>  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(777,3): error : Native linking failed. Please review the build log.

As before, if I remove your plugin, everything builds and works.
Could it be those missing symbols are in one of your dependencies? What version of said dependencies do you use?

Were you able to get try your build on a clean mac?

Have a nice weekend!
JR

**Comment by rdelrosario** [comment]
Not yet haven't had time. But will get back to you once I do.

**Comment by rdelrosario** [comment]
I tested it on other Mac and didn’t get any issues.

**Comment by rdelrosario** [comment]
Maybe could be that you don’t have all the Android SDKs up to date on the Android SDK Manager

**Comment by rdelrosario** [comment]
Unfortunately, haven't been able to reproduce this. So will have to close it for now. Did release a new version recently using the latest iOS FCM package just in case you want to try:

https://www.nuget.org/packages/Plugin.FirebasePushNotification/2.3.1-rc

**Comment by dais-set** [comment]
Thanks! Latest version seems to build ok, it was probably a mix of old xamarin versions and nuget silliness.
